tests/test_tiktoken.py

```diff
@@ -0,0 +1,139 @@
+    'hello <|endoftext|>', 'hello <|endoftext|> ', '<|endoftext}>',
+    '<|endoftext}> ', ' <|endoftext|>',
+    '<|endoftext|><|endoftext|><|endoftext|><|endoftext|>',
+    '<|endoftext|> <|endoftext|> <|endoftext|> <|endoftext|>'
+]
+
+TEST_STRINGS += HORRIBLE_STRINGS
+
+
+@pytest.mark.parametrize('model_name,encoding_name',
+                         [('gpt-4', None), ('gpt-3.5-turbo', None),
+                          ('text-davinci-003', None), (None, 'cl100k_base')])
+def test_tiktoken(model_name: Optional[str], encoding_name: Optional[str],
+                  tmp_path: pathlib.Path):
+    tiktoken = pytest.importorskip('tiktoken')
+
+    # Construction
+    wrapped_tokenizer = TiktokenTokenizerWrapper(model_name=model_name,
+                                                 encoding_name=encoding_name)
+    if model_name is not None:
+        original_tokenizer = tiktoken.encoding_for_model(model_name)
+    else:
+        original_tokenizer = tiktoken.get_encoding(encoding_name)
+
+    # Repr works
+    _ = wrapped_tokenizer.__repr__()
+
+    # Save and load
+    wrapped_tokenizer.save_pretrained(tmp_path)
+    reloaded_wrapped_tokenizer = transformers.AutoTokenizer.from_pretrained(
+        tmp_path, trust_remote_code=True)
+
+    didnt_match = []
+    # Simple tokenization test
+    for string in TEST_STRINGS:
+        wrapped_output = wrapped_tokenizer(string)
+        original_output = original_tokenizer.encode(string,
+                                                    allowed_special='all')
+        reloaded_wrapped_output = reloaded_wrapped_tokenizer(string)
+
+        try:
+            assert wrapped_output['input_ids'] == original_output
+            assert set(wrapped_output.keys()) == {'input_ids', 'attention_mask'}
+            assert reloaded_wrapped_output == wrapped_output
+        except:
+            assert False
+
+    # Round trip
+    for string in TEST_STRINGS:
+        wrapped_output = wrapped_tokenizer.decode(
+            wrapped_tokenizer(string)['input_ids'])
+        original_output = original_tokenizer.decode(
+            original_tokenizer.encode(string, allowed_special='all'))
+        reloaded_wrapped_output = reloaded_wrapped_tokenizer.decode(
+            reloaded_wrapped_tokenizer(string)['input_ids'])
+        assert wrapped_output == string
+        assert original_output == string
+        assert reloaded_wrapped_output == string
+
+    # Batched tokenization
+    wrapped_output = wrapped_tokenizer(
+        ['Hello world!', 'Hello world but longer!'])
+    original_output = original_tokenizer.encode_batch(
+        ['Hello world!', 'Hello world but longer!'])
+    reloaded_wrapped_output = reloaded_wrapped_tokenizer(
+        ['Hello world!', 'Hello world but longer!'])
+    assert wrapped_output['input_ids'] == original_output
+    assert set(wrapped_output.keys()) == {'input_ids', 'attention_mask'}
+    assert reloaded_wrapped_output == wrapped_output
```

Comment:
```suggestion
    assert 'input_ids' in wrapped_output and 'attention_mask' in wrapped_output
    assert wrapped_output['input_ids'] == original_output
    assert reloaded_wrapped_output == wrapped_output
```

Reply:
I actually want to test that there are no _other_ keys, so I will leave it as is.